midend/interpreter.cpp

```diff
@@ -310,33 +310,50 @@ void SymbolicStruct::dbprint(std::ostream& out) const {
 }
 
 SymbolicHeaderUnion::SymbolicHeaderUnion(const IR::Type_HeaderUnion* type,
-                               bool uninitialized,
-                               const SymbolicValueFactory* factory) :
-        SymbolicStruct(type, uninitialized, factory),
-        valid(new SymbolicBool(false)) {}
+                                         bool uninitialized,
+                                         const SymbolicValueFactory* factory) :
+        SymbolicStruct(type, uninitialized, factory) {}
+
+void SymbolicHeaderUnion::setValid(bool v, cstring field) {
+    CHECK_NULL(field);
+    fieldValue[field]->to<SymbolicHeader>()->setValid(v);
+    for (auto f : type->to<IR::Type_StructLike>()->fields) {
+        if (f->name.name != field)
+            fieldValue[f->name.name]->checkedTo<SymbolicHeader>()->setValid(false);
+    }
+}
 
-void SymbolicHeaderUnion::setValid(bool v) {
-    if (!v)
-        setAllUnknown();
-    valid = new SymbolicBool(v);
+SymbolicBool* SymbolicHeaderUnion::isValid() const {
```

Comment:
I fixed it

Reply:
The rule to me seems to be:
- if any field has the value "unknown" or "uninitialized", the result is "unknown"
- otherwise if exactly one field is valid the result SymblicBool(true), which is the really field->value.
 (it should not be possible for more than 1 to be true, that should be a BUG).
- otherwise if no field is valid the result is SymbolicBool(false).

The code below does not seem to do that, in particular I don't see the check for uninitialized or unknown.